api/drf_views.py

```diff
@@ -814,7 +816,7 @@ def get_serializer_class(self):
 class ProfileViewset(viewsets.ModelViewSet):
     serializer_class = ProfileSerializer
     authentication_classes = (TokenAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticated, DenyGuestUserMutationPermission)
```

Comment:
Let's check if IsAuthenticated is useful here.
```suggestion
    permission_classes = (DenyGuestUserMutationPermission)
```

Reply:
This should be fine. We do have IsAuthenticated baked in DenyGuestUserMutationPermission, but having IsAuthenticated adds a bit of documentations about the endpoint.